test/spreadPath.js

```diff
@@ -0,0 +1,45 @@
+import RA from '../src/index';
+import eq from './shared/eq';
+
+
+describe('spreadPath', function() {
```

Comment:
We should finally utilize the power of mocha more. I'll demonstrate what I have in mind.
Also don't try to create variables when they are not needed.

```javascript
import RA from '../src/index';
import eq from './shared/eq';


describe('spreadPath', function() {
  let obj;

  beforeEach(function () {
    obj =  {
      a: 1,
      b1: { b2: { c: 3, d: 4 } },
    };
  });

  it('tests currying', function () {
    const path = ['b1', 'b2'];
    const expected = { a: 1, c: 3, d: 4  };

    eq(RA.spreadPath(path, obj), expected);
    eq(RA.spreadPath(path)(obj), expected);
  });

  context('when path leads  to non object', function () {
    specify('should return object without path', function () {
      const obj = {
        a: 1,
        b1: { b2: 999 },
      };
      eq(RA.spreadPath(['b1', 'b2'], obj), { a: 1, b1: {} });    
    });
  });
  
  context("when path doesn't exist", function () {
    specify('should return object with identical structure as provided object', function () {
      eq(RA.spreadPath(['does', 'not', 'exist'], obj), obj);
    });
  });  
});